### DISCUSSION TITLE
Logging level in Debug window

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js-ui#1543] [Q&A]
Question about the debug window and logging for General and ZWave options in the CP.  

Do BOTH the settings in General and ZWave affect what is seen in the Debug window? It appears, to me, that General does, but ZWave doesn't. 

Do these parameters affect just the file logs, or do/should they affect the Debug window as well? 

I didn't see any clarification in the docs. Any advice would be appreciated.

**Comment by robertsLando** [comment]
What you see there is everything that is written to process.stdout/stderr of z2m. It means that you will see zwavejs2mqtt logs (based on the log level set on settings) and also zwavejs logs if log to file is disabled, otherwise zwavejs will log everything only to files in store directory

**Comment by GrizzlyAK** [comment]
Cool, thanks for the clarification
 Daniel. That last part was what I was missing. I had log to file set so was seeing only the zw2m logs.

**Comment by GrizzlyAK** [comment]
In the future, are you considering allowing for a filter on the Debug window (or just the zw2m log messages), such as particular Nodes like in the driver? I used tail | grep a LOT under the old OZW when I had direct access to the log files to see exactly what I wanted. But I don't have such access under zwavejs (on Home Assistant OS, anyway). 

My transition to zwavejs/zwavejs2mqtt in HA OS has been very smooth. I have over 500 entities that I had to rename so that all my automations still worked with only minimal changes (like where binary_sensors were now available whereas I had to create my own under OZW). And it seems much more responsive. The UI CP is very handy and is why I chose zw2m over the HA Add-On. Just still trying to get my head around the separation of duties among the various parts and how to navigate to do what I need to do. Thanks for your efforts on this project. Well done!
